CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
+| **settingsExample**        | An example configuration for the plugin's settings.                                                                                                                                                          | No       | `null`          |
+| **settingsDescription**    | A description of the settings provided to the `require('${luaName}')${setup}` function.                                                                                                                      | No       | `"Options provided to the require('${luaName}')${setup} function."` |
+| **hasSettings**            | Indicating whether the plugin has settings.                                                                                                                                                                  | No       | `true`          |
+| **extraOptions**           | Additional options for the plugin.                                                                                                                                                                           | No       | `{}`            |
+| **luaName**                | The Lua name for the plugin.                                                                                                                                                                                 | No       | `name` parameter |
+| **setup**                  | The setup function for the plugin.                                                                                                                                                                           | No       | `".setup"`      |
+| **extraConfig**            | Additional configuration for the plugin.                                                                                                                                                                     | No       | `{}`            |
+| **extraPlugins**           | Extra plugins to include.                                                                                                                                                                                    | No       | `[]`            |
+| **extraPackages**          | Extra packages to include.                                                                                                                                                                                   | No       | `[]`            |
+| **callSetup**              | Indicating whether to call the setup function.                                                                                                                                                               | No       | `true`          |
+| **installPackage**         | Indicating whether to install the package.                                                                                                                                                                   | No       | `true`          |
```

Comment:
Is this deprecated? What scenarios do we use this in?

Reply:
I haven't checked TBH. I only know the  `callSetup` which is used to tell when a lib is packaged (generally) and doesn't really on a setup function.

Reply:
Seems I added it back in https://github.com/nix-community/nixvim/commit/843fb302ebaa2559b7c50ad69ba9a00ae1a5836d, although it is a very niche option...

A quick search shows it is only used by `treesitter`: https://github.com/nix-community/nixvim/blob/6f210158b03b01a1fd44bf3968165e6da80635ce/plugins/by-name/treesitter/default.nix#L431-L434

Maybe add a caveat:
```suggestion
| **installPackage**         | Whether to install the package. This should almost never be disabled. It is occasionally disabled by plugin modules that need to apply additional overrides to the package before it is added to `extraPackages`. | No       | `true`          |
```

Reply:
Sorry, didn't see this PR. Thanks for catching that Matt. I remember you adding that specifically for treesitter way back when lol

Reply:
> Seems I added it back in https://github.com/nix-community/nixvim/commit/843fb302ebaa2559b7c50ad69ba9a00ae1a5836d, although it is a very niche option...
> 
> 
> 
> A quick search shows it is only used by `treesitter`: https://github.com/nix-community/nixvim/blob/6f210158b03b01a1fd44bf3968165e6da80635ce/plugins/by-name/treesitter/default.nix#L431-L434
> 
> 
> 
> Maybe add a caveat along the lines of?
> 
> ```suggestion
> 
> | **installPackage**         | Whether to install the package. This should almost never be disabled. It is occasionally disabled by plugin modules that need to apply additional overrides to the package before it is added to `extraPackages`. | No       | `true`          |
> 
> ```

@MattSturgeon Do you have something in mind? Without really knowing the full purpose of it, I find the line enough to either not mess with it, or start working on it.

Reply:
> Do you have something in mind?

What I had in mind was something along the lines of my earlier suggestion:

``Whether to install the package. This should almost never be disabled. It is occasionally disabled by plugin modules that need to apply additional overrides to the package before it is added to `extraPackages`.``

This is a little wordy though, I'll admit!

> Without really knowing the full purpose of it, I find the line enough to either not mess with it, or start working on it.

Yeah, I agree this is difficult. And we don't need to have everything perfect since things can always be improved down the line.

Reply:
I've made a small change to your sentence. It should be a little bit less verbose. But also a little bit less comprehensible. Lets see what you think about it :+1:

Reply:
See #2524